blog/2024-03-09-import_executor_default.md

```diff
@@ -0,0 +1,15 @@
+---
+author: J. Nick Koston
+authorURL: https://github.com/bdraco
+title: "Integrations import in the executor to avoid blocking the event loop"
+---
+
+Starting from Home Assistant 2024.4, all integrations are imported in the executor to avoid blocking the event loop while the code is imported and executed.
+
+In the rare event that an integration creates loop-bound objects at import time, this may fail because there is no running loop in the executor.
+
+Custom integrations needing more time to adjust their code to avoid creating such objects can opt-out and continue to have their code imported in the event loop, understanding that system stability will be impacted.
```

Comment:
If there's already a known release when the support for this option is going away, it's probably worth mentioning already in this blog post. Looks otherwise good to me :+1:

Reply:
No timeline for removing it yet. The plan is to wait until we see if anything breaks and than decide if we need a short or long timeline for removal. There isn't much incentive to rush it since the default is to not do blocking I/O and its likely better to give users an escape hatch if they can't get an old integration working otherwise as its pretty simple to add it to the manifest even if the cc author has abandoned the component.